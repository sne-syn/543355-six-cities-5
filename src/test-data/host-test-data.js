export const hostPro = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: true,
  name: `Janette`,
};
export const host = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: false,
  name: `Janette`,
};
