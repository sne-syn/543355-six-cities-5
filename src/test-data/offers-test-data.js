export const offers = [{
  bedrooms: 1,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Paris`,
  },
  description: `Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish`,
  goods: [`cabel TV`, `heating`, `kitchen`, `towels`, `washing machine`, `washer`, `dishwasher`, `conditioner`, `elevator`, `coffee machine`],
  host: {
    avatar: `https://robohash.org/28?set=set2&size=74x74`,
    id: 4,
    isPro: false,
    name: `Salom`,
  },
  id: 1,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/34LWdhj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/33Qpecj`, `https://bit.ly/30YVQif`, `https://bit.ly/36TgIv7`, `https://bit.ly/3lEt2DG`],
  isFavorite: true,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 8,
  price: 207,
  rating: 1.4018800815887484,
  title: `La Latina- Small Charming Studio`,
  type: `room`,
}, {
  bedrooms: 3,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Brussels`,
  },
  description: `In full nature and very close to public transport. Green area surrounded by all kinds of sports facilities including a golf course, paddle, artificial grass soccer, pelota court. In an urbanization of villas with landscaped plots of about 500 m2`,
  goods: [`elevator`, `dishwasher`, `coffee machine`, `washer`, `washing machine`],
  host: {
    avatar: `https://robohash.org/82?set=set2&size=74x74`,
    id: 6,
    isPro: false,
    name: `Whitney`,
  },
  id: 2,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/30YVQif`, `https://bit.ly/3lEt2DG`, `https://bit.ly/36TgIv7`, `https://bit.ly/33Qpecj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/34LWdhj`],
  isFavorite: false,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 5,
  price: 194,
  rating: 1.9069547983791004,
  title: `Estudio/suite completo en el centro de Segovia`,
  type: `house`,
}];
