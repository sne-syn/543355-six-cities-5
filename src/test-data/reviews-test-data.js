export const review = {
  author: `Beaulah`,
  avatar: `https://robohash.org/54?set=set2&size=54x54`,
  date: `2004-09-06T16:12:32.554Z`,
  id: 0,
  rating: 1,
  comment: `Antoni is a great host and us a wonderful home. His suggestions for places to go were killer!10/10 would stay again.`,
};

export const reviews = [{
  author: `Ione`,
  avatar: `https://robohash.org/98?set=set2&size=54x54`,
  date: `2007-10-13T16:12:32.554Z`,
  id: 1,
  rating: 4,
  comment: `There’s a stunning beach with great snorkelling just a 6min drive, a few other beautiful beaches close by too.`,
}, {
  author: `Bruno`,
  avatar: `https://robohash.org/53?set=set2&size=54x54`,
  date: `2005-08-25T16:12:32.554Z`,
  id: 2,
  rating: 3,
  comment: `Maria’s place is at a perfect location, very easy to get around with public transportation and walking distance to awesome bars and restaurants. Her place is great, very clean and has everything you need.`,
}];
