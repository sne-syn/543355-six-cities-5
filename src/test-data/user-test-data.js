export const user = {
  userAvatar: `https://robohash.org/82?set=set2&size=74x74`,
  userEmail: `trust_no_1@gmail.com`,
  userId: 1,
  userIsPro: false,
  userName: `fox_mulder`
};
