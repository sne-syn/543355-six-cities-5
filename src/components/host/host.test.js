import React from 'react';
import renderer from 'react-test-renderer';
import Host from './host';
import {host, hostPro} from '../../test-data/host-test-data';

const description = `Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish`;

describe(`<Host /> test`, () => {
  it(` should render Host correctly`, () => {
    const tree = renderer.create((
      <Host
        description={description}
        host={host}
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`should render Host with Pro-mark correctly`, () => {
    const tree = renderer.create((
      <Host
        description={description}
        host={hostPro}
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
