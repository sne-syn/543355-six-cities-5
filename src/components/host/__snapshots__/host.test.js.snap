// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Host /> test  should render Host correctly 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper "
    >
      <img
        alt="Host avatar"
        className="property__avatar user__avatar"
        height="74"
        src="/https://robohash.org/61?set=set2&size=74x74"
        width="74"
      />
    </div>
    <span
      className="property__user-name"
    >
      Janette
    </span>
  </div>
  <div
    className="property__description"
  >
    <p
      className="property__text"
    >
      Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish
    </p>
  </div>
</div>
`;

exports[`<Host /> test should render Host with Pro-mark correctly 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper property__avatar-wrapper--pro"
    >
      <img
        alt="Host avatar"
        className="property__avatar user__avatar"
        height="74"
        src="/https://robohash.org/61?set=set2&size=74x74"
        width="74"
      />
    </div>
    <span
      className="property__user-name"
    >
      Janette
    </span>
  </div>
  <div
    className="property__description"
  >
    <p
      className="property__text"
    >
      Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish
    </p>
  </div>
</div>
`;
